fedimg/utils.py

```diff
@@ -205,3 +206,39 @@ def get_image_name_from_ami_name_for_fedmsg(image_name):
     image_name = name_vt_region.rsplit('-', 4)[:-4][0]
 
     return image_name
+
+
+def cancel_stale_conversion_tasks():
+    _log.info("Preparing to cancel conversion tasks")
+    output, error, retcode = external_run_command([
+        'euca-describe-conversion-tasks --region us-east-1 | grep active',
+    ])
+
+    if retcode != 0:
+        return False
+
+    tasks = output.split('\n')[1:-1]
+
+    for task in tasks:
+        expiration_datetime = datetime.datetime.strptime(
+            task.split()[5],
+            '%Y-%m-%dT%H:%M:%Sz'
+        )
+        import_vol_task_id = task.split[3]
+
+        start_datetime = expiration_datetime - datetime.timedelta(days=3)
```

Comment:
Any specific reason to pick timedelta value as 3 days?
I am looking at some of the stale task of fedimg and it says expiration date as 2018-07-16 . If we consider timedelta as 3 days, then start_datetime of the task appears to be 2018-07-13 which is not really practical as today's date is 2018-07-11.

Reply:
yes it's very odd the API I think reports the time in which the task will be cleaned up maybe ? It is some future time. we need to find out what the time is set to and relate it back to the current time (i.e. create a task and immediately view the date returned from the API (should be something like **X days in the future** or something like that) and then figure out the time the task was created by subtracting X from the date reported from the API.

Reply:
When I was testing, I tested using a stale data which I had saved. This is a fault on my side, this should have been 7 days as @davdunc had confirmed.